trl/trainer/utils.py

```diff
@@ -52,9 +52,22 @@ def update(self, current, n_steps):
 
 
 class DataCollatorForCompletionOnlyLM(DataCollatorForLanguageModeling):
-    def __init__(self, response_template, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    """
+    Data collator used for completion tasks. It ensures that all the tokens of the labels are set to an 'ignore_index'
+     up to the prompt response template tokens ('response_template'). This ensure that the loss is only
+     calculated on the completion of the reponse.
+
+    Args:
+        response_template (`str`): the template form that indicates the start of the response, typically something like
+            '### Response:\n'
+        ignore_index (`int`, *optional*, defaults to `-100`):
+            The index to use to ignore the initial tokens with
+    """
+
+    def __init__(self, response_template: str, *args, ignore_index: int = -100, **kwargs):
+        super().__init__(*args, mlm=False, **kwargs)
```

Comment:
E.g. here we pass the `mlm` explicitly which is the reason the tests fail:
https://github.com/lvwerra/trl/blob/2b531b92238f9fc40c6cc9cf27b8bbd3ae2727b1/trl/trainer/sft_trainer.py#L170

We should check if mlm is a key in `kwargs` and only update with `False` if not.

Reply:
Doesn't that make it the same as the initial PR (not hardcoded) but just less transparent?

Original:

```
    def __init__(self, response_template: str, *args, mlm: bool = False, ignore_index: int = -100, **kwargs):
        super().__init__(*args, mlm=mlm, **kwargs)
```

Requested refactor:

```
    def __init__(self, response_template: str, *args, ignore_index: int = -100, **kwargs):
        super().__init__(*args, mlm=kwargs.get("mlm", False), **kwargs)
```

From a user/typing/IDE completion, I'd argue the first one is the better choice, no?

Reply:
True, let's go with the original one. Sorry for going in a circle to find the best solution :)

Reply:
No problem ;-) Re-added